internal/cmd/api/api.go

```diff
@@ -0,0 +1,332 @@
+			cl = oauth2.NewClient(ctx, oauth2.StaticTokenSource(tok))
+		} else if ch.Config.ServiceToken != "" && ch.Config.ServiceTokenID != "" {
+			req.Header.Set("Authorization", ch.Config.ServiceTokenID+":"+ch.Config.ServiceToken)
+			cl = &http.Client{}
+		}
+		res, err := cl.Do(req)
+		if err != nil {
+			return errors.Wrap(err, "sending HTTP request")
+		}
+		defer res.Body.Close()
+
+		if _, err := io.Copy(os.Stdout, res.Body); err != nil {
+			return errors.Wrap(err, "reading HTTP response body")
+		}
+
+		if res.StatusCode > 399 {
+			return errors.Errorf("HTTP %s", res.Status)
+		}
+
+		return nil
+	}
+
+	return cmd
+}
+
+func parseURL(ch *cmdutil.Helper, opts *ApiOpts, endpoint string) (*url.URL, error) {
+	reqPath := endpoint
+	reqPath = strings.ReplaceAll(reqPath, `{org}`, ch.Config.Organization)
+	reqPath = strings.ReplaceAll(reqPath, `{db}`, ch.Config.Database)
+	reqPath = strings.ReplaceAll(reqPath, `{branch}`, ch.Config.Branch)
+
+	u, err := url.Parse(ch.Config.BaseURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing base URL")
+	}
+	u = u.ResolveReference(&url.URL{Path: path.Join("v1", reqPath)})
+
+	if len(opts.Query) > 0 {
+		for _, param := range opts.Query {
+			k, v, ok := strings.Cut(param, "=")
+			if !ok {
+				return nil, errors.Wrapf(err, "parsing query param %q", param)
+			}
+			q := u.Query()
+			q.Add(k, v)
+			u.RawQuery = q.Encode()
+		}
+	}
+	return u, nil
+}
+
+func parseHeader(opts *ApiOpts, method, userAgent string, defaultHeaders map[string]string) (http.Header, error) {
+	out := make(http.Header)
+	for k, v := range defaultHeaders {
+		out.Set(k, v)
+	}
+	out.Set("User-Agent", userAgent)
+	switch method {
+	case http.MethodGet:
+		out.Set("Accept", "application/json")
+	case http.MethodPost, http.MethodPut, http.MethodPatch:
+		out.Set("Accept", "application/json")
+		out.Set("Content-Type", "application/json")
+	}
+	for _, header := range opts.Header {
+		k, v, ok := strings.Cut(header, ":")
+		if !ok {
+			return nil, errors.Errorf("invalid header: %q", header)
+		}
+		out.Set(k, strings.TrimPrefix(v, " "))
+	}
+	return out, nil
+}
+
+func parseBody(opts *ApiOpts) (io.Reader, error) {
+	if opts.Input == "" && len(opts.Field) == 0 {
+		return nil, nil
+	}
+
+	var (
+		bodyMap map[string]interface{}
+		raw     []byte
+		err     error
+	)
+	if opts.Input != "" {
+		if opts.Input == "-" {
+			raw, err = io.ReadAll(os.Stdin)
+			if err != nil {
+				return nil, errors.Wrap(err, "reading body from stdin")
+			}
+		} else {
+			raw, err = os.ReadFile(opts.Input)
+			if err != nil {
+				return nil, errors.Wrapf(err, "reading body from file %q", opts.Input)
+			}
+		}
+	}
+	if len(raw) > 0 {
+		bodyMap = make(map[string]interface{})
+		if err := json.Unmarshal(raw, &bodyMap); err != nil {
+			// body wasn't JSON
+			if len(opts.Field) > 0 {
+				return nil, errors.Wrap(err, "parsing JSON from --input because --field specified (which requires JSON input)")
```

Comment:
I didn't understood this sentence. When does this happen?

Reply:
If you specify `--input` as a file, and the content of the file isn't JSON, but you also provided `--field` which implies you're trying to template the `--input` with `--field`. But you can't template non-JSON input, so if the input isn't isn't valid JSON and you passed `--field`, we return an error.